OSDOCS-3233: Rel Note for Azure minting removal (#42692)
openshift/openshift-docs: release_notes/ocp-4-10-release-notes.adoc

You cannot upgrade {op-system-base} 7 compute machines to {op-system-base} 8. You must deploy new {op-system-base} 8 hosts, and the old {op-system-base} 7 hosts must be removed.

[id="ocp-4-10-removed-feature-azure-mint-mode"]
==== Support for minting credentials for Microsoft Azure removed

> Support for using the Cloud Credential Operator (CCO) in mint mode on Microsoft Azure clusters has been removed. This change is due to the planned link:https://azure.microsoft.com/en-us/updates/update-your-apps-to-use-microsoft-graph-before-30-june-2022/[retirement of the Azure AD Graph API by Microsoft on 30 June 2022], and is being backported to all supported versions of {product-title} in z-stream updates.

Reviewer: ```suggestion
Support for using the Cloud Credential Operator (CCO) in mint mode on Microsoft Azure clusters has been removed. This change is due to the planned link:https://azure.microsoft.com/en-us/updates/update-your-apps-to-use-microsoft-graph-before-30-june-2022/[retirement of the Azure AD Graph API by Microsoft on 30 June 2022] and is being backported to all supported versions of {product-title} in z-stream updates.
```

For previously installed Azure clusters that use mint mode, the CCO attempts to update existing secrets. If a secret contains the credentials of previously minted app registration service principals, it is updated with the contents of the secret in `kube-system/azure-credentials`. This behavior is similar to passthrough mode.

[NOTE]
====

> In addition to the `Contributor` role that is required by mint mode, the app registration service principals require the `User Access Administrator` role that is used for passthrough mode.

Reviewer: I feel like this is missing something. Maybe:
```suggestion
In addition to the `Contributor` role that is required by mint mode, the modified app registration service principals now require the `User Access Administrator` role that is used for passthrough mode.
```

====

> While the Azure AD Graph API is still available, the CCO attempts to clean up previously minted app registration service principals.

Reviewer: So, does this mean that users could/should do something that is not the next few steps to avoid their app registration service principals getting orphaned in July?

> If the cluster is upgraded to a version of {product-title} that no longer supports mint mode after the Azure AD Graph API is retired, the CCO sets an `OrphanedCloudResource` condition on the associated `CredentialsRequest`, but does not treat the error as fatal. The condition includes a message similar to `unable to clean up App Registration / Service Principal: <app_registration_name>`. Cleanup after the Azure AD Graph API is retired requires manual intervention using the Azure CLI tool or the Azure web console to remove any remaining app registration service principals.

Reviewer: ```suggestion
If the cluster is upgraded to a version of {product-title} that no longer supports mint mode after the Azure AD Graph API is retired, the CCO sets an `OrphanedCloudResource` condition on the associated `CredentialsRequest` but does not treat the error as fatal. The condition includes a message similar to `unable to clean up App Registration / Service Principal: <app_registration_name>`. Cleanup after the Azure AD Graph API is retired requires manual intervention using the Azure CLI tool or the Azure web console to remove any remaining app registration service principals.
```

To clean up resources manually, you must find and delete the affected resources.

> . Using the Azure CLI tool, filter the app registration service principals using the `<app_registration_name>` from an `OrphanedCloudResource` condition message by typing the following command:

Reviewer: ```suggestion
. Using the Azure CLI tool, filter the app registration service principals that use the `<app_registration_name>` from an `OrphanedCloudResource` condition message by running the following command:
```

+
[source,terminal]
----
$ az ad app list --filter "displayname eq '<app_registration_name>'" --query '[].objectId'
----
+
.Example output
+
[source,terminal]
----
[
  "038c2538-7c40-49f5-abe5-f59c59c29244"
]
----

> . Delete the app registration service principal by typing the following command:

Reviewer: ```suggestion
. Delete the app registration service principal by running the following command:
```

+
[source,terminal]
----
$ az ad app delete --id 038c2538-7c40-49f5-abe5-f59c59c29244
----

[NOTE]
====
After cleaning up resources manually, the `OrphanedCloudResource` condition persists because the CCO cannot verify that the resources were cleaned up.
====

> 

Reviewer: Are users going to have to configure a different kind of CCO operation after they fix their orphaned resources?

Reviewer 2: If operator credentials mode is set to default i.e `""`, then the CCO will automatically convert it to passthrough from manual, but if it is set to `"Manual"` then user has to manually switch it to `"Passthrough"` or else CCO will throw error.

Author: > but if it is set to `"Manual"` then user has to manually switch it to `"Passthrough"` or else CCO will throw error.

@akhil-rane - do you mean if it is set to `"Mint"` here?

Reviewer 2: yeah sorry its `"Mint"`
